### PR TITLE
Added current and max swap and memory usage stats

### PR DESCRIPTION
**Pull request by Zyqsempai**
Related to #104 
Added `memory.current, memory.max, memory.swap.current, memory.swap.max` stats.
Signed-off-by: bpopovschi <zyqsempai@mail.ru>

### CODE REVIEW

**Comment by Zyqsempai** [comment]
@AkihiroSuda @crosbymichael PTAL

**Comment by estesp** [comment]
I think you will need to rebase on #118 when it is reviewed and merged to get CI passing for all your PRs.

File: v2/stats/metrics.proto [hunk 1]
```diff
@@ -56,6 +56,15 @@ message MemoryStat {
...
 	uint64 thp_collapse_alloc = 31;
+	MemoryEntry usage = 32;
+	MemoryEntry swap_usage = 33;
+}
+
+message MemoryEntry {
+	uint64 limit = 1;
+	uint64 usage = 2;
+	uint64 max = 3;
+	uint64 failcnt = 4;
```
Comment (AkihiroSuda) [inline v2/stats/metrics.proto]: we can omit unused fields

File: v2/stats/metrics.proto [hunk 2]
```diff
@@ -56,6 +56,15 @@ message MemoryStat {
 	uint64 pglazyfreed = 29;
 	uint64 thp_fault_alloc = 30;
 	uint64 thp_collapse_alloc = 31;
+	MemoryEntry usage = 32;
```
Comment (AkihiroSuda) [inline v2/stats/metrics.proto]: wondering this should be = 1

Comment (Zyqsempai) [reply v2/stats/metrics.proto]: What do you mean?

Comment (crosbymichael) [reply v2/stats/metrics.proto]: no

Comment (crosbymichael) [reply v2/stats/metrics.proto]: instead of a struct, can we just add these as field? keep the stats flat and simple

File: v2/manager.go [hunk 3]
```diff
@@ -326,6 +326,22 @@ func (c *Manager) Stat() (*stats.Metrics, error) {
 		ThpFaultAlloc:         out["thp_fault_alloc"].(uint64),
 		ThpCollapseAlloc:      out["thp_collapse_alloc"].(uint64),
 	}
+	cur_mem := getStatFileContent(filepath.Join(c.path, "memory.current"))
+	max_mem := getStatFileContent(filepath.Join(c.path, "memory.max"))
+	cur_swap := getStatFileContent(filepath.Join(c.path, "memory.swap.current"))
+	max_swap := getStatFileContent(filepath.Join(c.path, "memory.swap.max"))
+
+	usage := stats.MemoryEntry{
+		Usage: cur_mem,
```
Comment (crosbymichael) [inline v2/manager.go]: try not to use snake case `_` in the go code.

**Comment by crosbymichael** [comment]
Can you rebase on master as well to fix the build?  Thanks!

File: v2/manager.go (hunk 3 above)
Comment (Zyqsempai) [reply v2/manager.go]: oh, yeah, sorry, Python legacy;)

File: v2/stats/metrics.proto (hunk 2 above)
Comment (Zyqsempai) [reply v2/stats/metrics.proto]: Sure

**Comment by Zyqsempai** [comment]
@crosbymichael Done.

File: v2/utils.go [hunk 4]
```diff
@@ -216,3 +218,24 @@ func ToResources(spec *specs.LinuxResources) *Resources {
 	}
 	return &resources
 }
+
+// Gets uint64 parsed content of single value cgroup stat file
+func getStatFileContent(filePath string) uint64 {
```
Comment (fuweid) [inline v2/utils.go]: nit: function name is not specific, but it only can retrieve uint64 value.

Comment (AkihiroSuda) [reply v2/utils.go]: Also, this should take `io.Reader` rather than a filepath string and should have unit tests. Can be a follow-up PR.

Comment (Zyqsempai) [reply v2/utils.go]: Yep, i have in plans to cover everything with unit tests but in separate PR's

Comment (Zyqsempai) [reply v2/utils.go]: Also, I think that `io.Reader` parameter here will over complicate using of that function.

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**